### PR TITLE
Fixed teeth pill overload by adding a teeth count to species

### LINKED ISSUE #83820: Entire pharmacy in your mouth
Reporting client version: 515.1640

## Issue Summary

You can have seemingly infinite amount of pills in your mouth!

Discovered by me and Sherman Cottle
## Testmerges:
- [[TM ONLY] Hijacked Revision Date for v3 TGS Api](https://www.github.com/tgstation/tgstation/pull/83227)
- [Incoming stamina damage while in stamcrit has diminishing returns applied](https://www.github.com/tgstation/tgstation/pull/83445)
- [Cult Vs. Heretic: 7 Months Later Edition](https://www.github.com/tgstation/tgstation/pull/82877)
- [Adds some logging to roundstart, to figure out why it's been taking longer recently](https://www.github.com/tgstation/tgstation/pull/83745)

## Reproduction:

Drapes
Mouth
Drill
Insert pill
Repeat

image

Please cap it at like... 6 or 4.

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

There's now a limit to how many dental implants you can cram into your mouth, which is governed by your species' teeth limit.

Most species have 32 teeth, due to being based on humans, weirding me out when thinking about their teeth, or lack of enough information

Moths and flypeople have NO teeth. They CAN'T get dental implants. I'm NERFING moths.

Lizards have seventy-five (!!!) teeth. Lizards are weird.

## Why It's Good For The Game

> There's now a limit to how many dental implants you can cram into your mouth, which is governed by your species' teeth limit.

Closes #83820 by, uh. Well I guess it doesn't really do anything about being able to have 32 pills in your teeth... but now it's realistically limited. Yay! Anyone willing to make someone else suffer through 32 dental surgeries deserves it.

> Most species have 32 teeth, due to being based on humans, weirding me out when thinking about their teeth, or lack of enough information

Bro do slimegirls have teeth? Podpeople? Zombies?? Ayys????

> Moths and flypeople have NO teeth. They CAN'T get dental implants. I'm NERFING moths.

Moths and flies DON'T have teeth. It's UNREALISTIC. And it's a small and funny touch, balanced out by...

> Lizards have seventy-five (!!!) teeth. Lizards are weird.

Lizards are weird. 

## Changelog

:cl:
fix: There's now a limit to how many dental implants you can cram into your mouth, which is governed by your species' teeth limit.
refactor: Most species have 32 teeth, due to being based on humans, weirding me out when thinking about their teeth, or lack of enough information
refactor: Moths and flypeople have NO teeth. They CAN'T get dental implants. I'm NERFING moths.
refactor: Lizards have seventy-five (!!!) teeth. Lizards are weird.
/:cl:

### CODE REVIEW

**Comment by MrMelbert** [comment]
75 (odd number) feels weird, like which row has 1 less tooth for no reason? 
Maybe it round it down to 72

File: code/modules/mob/living/carbon/human/species_types/lizardpeople.dm
Comment (carlarctg) [inline code/modules/mob/living/carbon/human/species_types/lizardpeople.dm:29]:
```suggestion
	teeth_count = 72
```

**Comment by CCC23** [comment]
you can put more than 1 tooth implant in?

**Comment by FlufflesTheDog** [comment]
I have to imagine that non-molars would make tooth pills very awkward

**Comment by MrMelbert** [comment]
> I have to imagine that non-molars would make tooth pills very awkward

quitter talk

**Comment by Deadgebert** [comment]
why? why nerf moth?

**Comment by YuliceTheDruid** [comment]
Straub Nerf 3 years in the making.

**Comment by rasepretep** [comment]
Please tell me that the odd-numbered tooth in lizards is a hidden reference to egg-tooth, a tooth-like protrusion on the top of the snout in reptiles and birds, that allows hatchlings to break out of the egg from the inside.

**Comment by nianjiilical** [comment]
snail people should technically have thousands of teeth when you think about it, have you ever seen those motherfuckers crunching lettuce

also should slimepeople have teeth? i mean i guess you can technically put a dental implant in a gummy tooth

**Comment by Formerxboxgamer** [comment]
It’s lame to lock out an entire species from an aspect of the game like this. Let moths implant pills in their gums or something I don’t know.

**Comment by norsvenska** [comment]
moths dont have teeth?
explain THIS then!
image

**Comment by carlarctg** [comment]
> It’s lame to lock out an entire species from an aspect of the game like this. Let moths implant pills in their gums or something I don’t know.

no

**Comment by carlarctg** [comment]
> moths dont have teeth? explain THIS then! image

AAAAAAAAAAAAAAAAAAAAAAOOOOOOOOOOOOOOOOEEEEEEEEEEEEEEEEEUUUUUUUUUUUUUURGGGGGGGGGGGHHHHHHHHH

**Comment by rasepretep** [comment]
> moths dont have teeth? explain THIS then! image

Dentures. That'll be a followup PR

**Review by ZephyrTFA — APPROVED** [review]
I hate this pr.

**Comment by nikothedude** [comment]
we are fianlly getting closer to spitting teeth out when we're punched in the mouth

all we need is to fully canonize teeth as seventy two instances of the teeth datum in its own special organ slot and we're set

**Comment by IndieanaJones** [comment]
> we are fianlly getting closer to spitting teeth out when we're punched in the mouth
> 
> all we need is to fully canonize teeth as seventy two instances of the teeth datum in its own special organ slot and we're set

You're thinking too small, every individual tooth should be its own organ :)

**Comment by ZephyrTFA** [comment]
> > we are fianlly getting closer to spitting teeth out when we're punched in the mouth
> > 
> > all we need is to fully canonize teeth as seventy two instances of the teeth datum in its own special organ slot and we're set
> 
> You're thinking too small, every individual tooth should be its own organ :)

that's a git ban

**Comment by Deultus** [comment]
now add the ability to knock them out like organs, aiming at the mouth. It will be such an end

**Comment by DoomAngelBlade** [comment]
This pr fixes nothing.
This pr is garbage =(

Problems with having alot of tooth pills are:
Possibly endless suply of meds on you at all times(32 * 50  = Shitton. You can literally give yourself temp. blood, regen, teleport each time you're hit/wounded)
Your're handicapping yourself by removing half/quarter of your screen.
If youre in spess suit / mod suit - only ways to apply meds are - patches(takes time, stand still), autoinjectors(only certain meds, hard to produce, is limited) With a pharmacy in your mouth - it negates that and for an absurdly long time.

Setting limit to 4-6 pills would be very good, imo.
It will not throw the possibilities of two people who know chemistry AND surgery + have time for it to make any fight in space offset to one who has most tooth pills.

It will still be advantageous to have tooth pills but not overpowered.
( I mean, it requires a chemist and 2 peopel to do it cause you can't self implant it)

OR
Atleast make you have like (( (amount of pills) * (amount of pills) ) /2 - 10 )%, min 0 chance to consume / pop a random pill when hit in mouth and (( (amount of pills) * (amount of pills) ) /4 - 10 )% , min 0 when hit in head with anything that does brute.
4 pills = (Mouth) 0 % / 0 % (Head) chance to pop a pill
5 pills = 2.5% / 0%  
6 pills = 8% / 0%
7 pills = 14.5% / 2.25%
8 pills = 22% /  6%

etc.
Will add a funny mechanic that if you're gonna get hurt with a punch and you hav alot of teleport / small healing pills - you will start getting healed by yourself.

Or just OD on all the meds after 3 punches because you have not teeth but a big bunch of capsules in your mouth that you suddenly chewed on.

**Comment by carlarctg** [comment]
Mister overcomplication above

Its fine to have 32 pills because that means you need to prepare 32 pills full of stimulants and medicine AND you need to pester someone to do surgery on you 32 times

Anything else is beyond stupid as its trying to address a problem that doesnt exist.

**Comment by Higgin** [comment]
Explains why they like soups and soft foods so much

Need the follow-up PR that makes the moth eating sounds just go "nom nom nom" 

(this is a hilarious and good PR.)

**Review by optimumtact — APPROVED** [review]

**Comment by optimumtact** [comment]
You need maintainer approval for species changes so you should find someone to approve it.

**Comment by Maurukas** [comment]
This is just a backdoor way of adding dentist as a new job.

**Comment by Loafin34** [comment]
add in knocking out teeth since we're doing teeth

**Comment by GPeckman** [comment]
If you're gonna do the teeth thing, please tie it to a character's head, not the species datum. We're trying to move away from behavior directly tied to species.

**Comment by carlarctg** [comment]
> If you're gonna do the teeth thing, please tie it to a character's head, not the species datum. We're trying to move away from behavior directly tied to species.

nerd voice (ok

**Comment by Jacquerel** [comment]
Drafting until it is on head and not species

**Pull request merged**
